Sources/Logging/TestingLogHandler.swift

```diff
@@ -0,0 +1,298 @@
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
+        return dateFormatter
+    }
+
+    public let timestamp: Date
+    public let label: String
+
+    public let level: Logger.Level
+    public let message: Logger.Message
+    public let metadata: Logger.Metadata
+    public let source: String
+
+    public let file: String
+    public let function: String
+    public let line: UInt
+
+    public var prettyMetadata: String {
+        metadata.sorted(by: {
+            $0.key < $1.key
+        }).map {
+            "\($0)=\($1)"
+        }.joined(separator: " ")
+    }
+
+    public var description: String {
+        "\(Self.timestampFormat.string(from: timestamp)) \(level) \(label) :\(prettyMetadata) [\(source)] \(message)"
+    }
+
+    public init(
+        timestamp: Date = Date(),
+        label: String = "nil",
+        level: Logger.Level = .info,
+        message: Logger.Message = .init(stringLiteral: ""),
+        metadata: Logger.Metadata = [:],
+        source: String = "nil",
+        file: String = "nil",
+        function: String = "nil",
+        line: UInt = 0
+    ) {
+        self.timestamp = timestamp
+        self.label = label
+        self.level = level
+        self.message = message
+        self.metadata = metadata
+        self.source = source
+        self.file = file
+        self.function = function
+        self.line = line
+    }
+
+    // Easily check if log message matches certain characteristics:
+    @available(macOS 13.0, *)
+    public func match(
+        label: String? = nil,
+        level: Logger.Level? = nil,
+        message: Regex<AnyRegexOutput>? = nil,
+        metadata: [(Logger.Metadata.Element) -> Bool] = [],
+        source: Regex<AnyRegexOutput>? = nil,
+        file: Regex<AnyRegexOutput>? = nil,
+        function: String? = nil,
+        line: UInt? = nil
+    ) -> Bool {
+        if let label,
+           self.label != label
+        {
+            return false
+        }
+        if let level,
+           self.level != level
+        {
+            return false
+        }
+        if let message,
+           !self.message.description.contains(message)
+        {
+            return false
+        }
+        for predicate in metadata {
+            if !self.metadata.contains(where: predicate) {
+                return false
+            }
+        }
+        if let source,
+           !self.source.contains(source)
+        {
+            return false
+        }
+        if let file,
+           !self.file.contains(file)
+        {
+            return false
+        }
+        if let function,
+           self.function != function
+        {
+            return false
+        }
+        if let line,
+           self.line != line
+        {
+            return false
+        }
+
+        return true
+    }
+}
+
+public struct TestingLogHandler: LogHandler, Sendable {
```

Comment:
Naming NIT to align it with `TestMetrics` from `swift-metrics`
```suggestion
public struct TestLogHandler: LogHandler, Sendable {
```